### PR TITLE
fix: MaterialOutlinedPasswordBox content shift

### PR DESCRIPTION
**Pull request by Kunal22shah**
﻿GitHub Issue: #1493

## PR Type
image

What kind of change does this PR introduce?

- Bugfix
- Feature
- Code style update (formatting)
- Refactoring (no functional changes, no api changes)
- Build or CI related changes
- Documentation content changes
- Other... Please describe:

## PR Checklist 
Please check if your PR fulfills the following requirements:

- Commits must be following the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [ ] Tested the changes where applicable:
	- [ ] UWP
	- [ ] iOS
	- [ ] Android
	- [ ] WASM
	- [ ] MacOS
- [ ] Updated the documentation as needed:
	- [ ] [General Doc Update](https://github.com/unoplatform/Uno.Themes/tree/master/doc)
	- [ ] [material-controls-styles.md](https://github.com/unoplatform/Uno.Themes/blob/master/doc/material-controls-styles.md)
	- [ ] [cupertino-controls-styles.md](https://github.com/unoplatform/Uno.Themes/blob/master/doc/cupertino-controls-styles.md)
	- [ ] [lightweight-styling.md (LightWeight Styling Resource Keys)](https://github.com/unoplatform/Uno.Themes/blob/master/doc/lightweight-styling.md)
- [ ] Contains **No** breaking changes
  > If the pull request contains breaking changes, commit message must contain a detailed description of the action to take for the consumer of this library. As explained by the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

File: src/library/Uno.Material/Styles/Controls/v2/PasswordBox.xaml [hunk 1]
```diff
@@ -471,32 +471,26 @@
 		<Setter Property="Template">
 			<Setter.Value>
 				<ControlTemplate TargetType="PasswordBox">
-					<Border x:Name="RootBorder"
-							BorderBrush="{TemplateBinding BorderBrush}"
-							BorderThickness="{TemplateBinding BorderThickness}"
-							CornerRadius="{TemplateBinding CornerRadius}"
-							Padding="{ThemeResource OutlinedPasswordBoxBorderPadding}">
```
Comment (Xiaoy312) [inline src/library/Uno.Material/Styles/Controls/v2/PasswordBox.xaml:478]: resources whose references are all removed, should also be removed
this change also need to be reflected, doc and c#markup

File: src/library/Uno.Material/Styles/Controls/v2/PasswordBox.xaml [hunk 2]
```diff
@@ -629,7 +628,12 @@
 									Style="{StaticResource MaterialRevealButtonStyle}"
 									Visibility="Collapsed" />
 						</Grid>
-					</Border>
+
+						<Border x:Name="BorderBorder"
```
Comment (Xiaoy312) [inline src/library/Uno.Material/Styles/Controls/v2/PasswordBox.xaml:632]: it is important, to comment, when you are using an esoteric setup
copy the comment i left on `BorderBorder` in `MaterialOutlinedTextBoxStyle`

**Review by Xiaoy312 — COMMENTED** [review]
you have introduced changes that needs to be documented in the migration docs

File: doc/material-migration.md [hunk 3]
```diff
@@ -29,6 +29,17 @@ We have standardized the naming of styles and resource keys. This change aims to
...
+#### Removal of PasswordBox Border Padding Resources
+
+The following resource keys related to `PasswordBox` border padding have been removed:
+
+| Old Resource Key | New Resource Key | Value |
+| ---- | ---- |---- |
+| OutlinedPasswordBoxBorderPadding | ***REMOVED*** | 1 |
+| OutlinedPasswordBoxBorderPaddingPointerOver | ***REMOVED*** | 0 |
+| OutlinedPasswordBoxBorderPaddingFocused | ***REMOVED*** | 0 |
+| OutlinedPasswordBoxBorderThicknessPointerOver | OutlinedPasswordBoxBorderThicknessPointerOver | 1 |
```
Comment (agneszitte) [inline doc/material-migration.md:41]: @Xiaoy312 I think the overall upgrading section should be reviewed/adjusted a bit more based on previous changes:
image
What do you think ?

Comment (Xiaoy312) [reply doc/material-migration.md:41]: It should be merged together.

File: doc/material-migration.md [hunk 4]
```diff
@@ -14,20 +14,23 @@ We have standardized the naming of styles and resource keys. This change aims to
...
+| MaterialFilledTextBoxBorderHeightFocused          | FilledTextBoxBorderHeightFocused                |
+| MaterialOutlinedTextBoxBorderThickness            | OutlinedTextBoxBorderThickness                  |
+| MaterialOutlinedTextBoxCornerRadius               | OutlinedTextBoxCornerRadius                     |
+| MaterialOutlinedTextBoxPadding                    | OutlinedTextBoxPadding                          |
+| MaterialOutlinedTextBoxMinHeight                  | OutlinedTextBoxMinHeight                        |
+| MaterialOutlinedTextBoxBorderThicknessFocused     | OutlinedTextBoxBorderThicknessFocused           |
+| MaterialOutlinedTextBoxBorderThicknessPointerOver | OutlinedTextBoxBorderThicknessPointerOver       |
+| OutlinedPasswordBoxBorderPadding                  | ***REMOVED***                                   |
+| OutlinedPasswordBoxBorderPaddingPointerOver       | ***REMOVED***                                   |
+| OutlinedPasswordBoxBorderPaddingFocused           | ***REMOVED***                                   |
```
Comment (agneszitte) [inline doc/material-migration.md:33]: Missing OutlinedPasswordBoxBorderThicknessPointerOver change @Xiaoy312 no?

File: doc/material-migration.md [hunk 5]
```diff
@@ -14,20 +14,23 @@ We have standardized the naming of styles and resource keys. This change aims to
```
Comment (agneszitte) [inline doc/material-migration.md:14]: Maybe this text can be adjusted a bit now I think @Xiaoy312 
image

File: doc/material-migration.md (hunk 4 above)
Comment (Kunal22shah) [reply doc/material-migration.md:33]: Do we consider `OutlinedPasswordBoxBorderThicknessPointerOver` value change as a breaking change? it is already mentioned in here doc/styles/PasswordBox.md

Comment (Xiaoy312) [reply doc/material-migration.md:33]: values change are non-breaking
plus we've made bigger change than throughout the year, and there was absolute 0 mention neither

File: doc/material-migration.md (hunk 5 above)
Comment (Xiaoy312) [reply doc/material-migration.md:14]: i think it is fine.

Comment (agneszitte) [reply doc/material-migration.md:14]: After another look, i am fine with it also now

**Review by Xiaoy312 — APPROVED** [review]

**Pull request merged**